extensions/wikia/EditPageLayout/js/plugins/WikitextSyntaxHighlighter.js

```diff
@@ -451,90 +416,159 @@ define('WikiTextSyntaxHighlighter', ['wikia.window', 'wikia.document', 'wikia.lo
+		wpTextbox1.style.borderLeftStyle = wpTextbox1Style.borderLeftStyle;
+		wpTextbox1.style.borderLeftWidth = wpTextbox1Style.borderLeftWidth;
+		wpTextbox1.style.borderRightStyle = wpTextbox1Style.borderRightStyle;
+		wpTextbox1.style.borderRightWidth = wpTextbox1Style.borderRightWidth;
+		wpTextbox1.style.borderTopLeftRadius = wpTextbox1Style.borderTopLeftRadius;
+		wpTextbox1.style.borderTopRightRadius = wpTextbox1Style.borderTopRightRadius;
+		wpTextbox1.style.borderTopStyle = wpTextbox1Style.borderTopStyle;
+		wpTextbox1.style.borderTopWidth = wpTextbox1Style.borderTopWidth;
+		wpTextbox1.style.boxSizing = "border-box";
+		wpTextbox1.style.clear = wpTextbox1Style.clear;
+		wpTextbox1.style.color = syntaxHighlighterConfig.foregroundColor;
+		wpTextbox1.style.fontFamily = wpTextbox1Style.fontFamily;
+		wpTextbox1.style.fontSize = wpTextbox1Style.fontSize;
+		wpTextbox1.style.lineHeight = "normal";
+		wpTextbox1.style.marginBottom = wpTextbox1Style.marginBottom; //lock to pixel value because the top margin was also locked to a pixel value when it was moved to wpTextbox0
+		wpTextbox1.style.marginLeft = "0";
+		wpTextbox1.style.marginRight = "0";
+		wpTextbox1.style.overflowX = "auto";
+		wpTextbox1.style.overflowY = "scroll";
+		wpTextbox1.style.padding = "0";
+		wpTextbox1.style.resize = resize;
+		wpTextbox1.style.tabSize = wpTextbox1Style.tabSize;
+		wpTextbox1.style.whiteSpace = "pre-wrap";
+		wpTextbox1.style.width = "100%";
+		wpTextbox1.style.wordWrap = "normal"; //overall more visually appealing
+
+		//lock both heights to pixel values so that the browser zoom feature works better
+		wpTextbox1.style.height = wpTextbox0.style.height = wpTextbox1.offsetHeight + "px";
+
+		//insert wpTextbox0 underneath wpTextbox1
+		wpTextbox1.style.marginTop = -wpTextbox1.offsetHeight + "px";
+		wpTextbox1.parentNode.insertBefore(wpTextbox0, wpTextbox1);
 
 		document.head.appendChild(syntaxStyleElement);
-		$(wpTextbox1).on('keydown', debouncedHighlightSyntax);
-		wpTextbox1.addEventListener('scroll', syncScrollX);
-		wpTextbox1.addEventListener('scroll', syncScrollY);
-		highlightSyntaxIfNeededIntervalId = setInterval(highlightSyntaxIfNeeded, 500);
-		highlightSyntax();
 
-		initialized = true;
+		wpTextbox1.addEventListener("input", highlightSyntax);
+		wpTextbox1.addEventListener("scroll", syncScrollX);
+		wpTextbox1.addEventListener("scroll", syncScrollY);
+		attributeObserver = new MutationObserver(syncTextDirection);
+		attributeObserver.observe(wpTextbox1, {attributes: true});
+		parentObserver = new MutationObserver(syncParent);
+		parentObserver.observe(wpTextbox1.parentNode, {childList: true});
+		highlightSyntaxIfNeededIntervalID = setInterval(highlightSyntaxIfNeeded, 500);
+		highlightSyntax();
 	}
 
-	function queueSetup (textarea) {
+	function queueSetup(textarea) {
 		setTimeout(function () {
 			setup(textarea);
 		}, 0);
 	}
 
-	function init (textarea, config) {
+	function init(textarea, config) {
```

Comment:
this function and it's args was added for backward compact and to allow exporting